.github/workflows/check.yaml

```diff
@@ -15,6 +11,11 @@ on:
       - '**.md'
       - '**.rst'
 
+concurrency:
+      # We do not want to run multiple jobs for single PR.
+      group: ${{ github.head_ref || github.ref_name }}
```

Comment:
This would cancel other workflows, e.g. documentation check, on the same branch, which is not correct. We  can use something like [this](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow) instead.

Reply:
You can check the `head_ref` [here](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context).

Reply:
We can use `${{ github.head_ref || github.ref_name }}-lint-unit-func`, but there is no other workflow with same group, so that's why I think it's ok like this.

Reply:
I understand `head_ref` is for each PR only, but within a PR we could have more workflows triggered. I still prefer using `${{ github.workflow }}-${{ github.ref }}` for each workflow to make sure the groups are tied to a single workflow definition to avoid future overhead. And I think it would be a good idea to have a team-wide template/standard for this kind of things.

Reply:
`${{ github.workflow }}-${{ github.head_ref || github.ref }}` should be good, thanks.